libgit2: update to 1.7.1; py-pygit2: update to 1.13.0; rev-bump all dependents

mascguy opened the pull request:
### Description

Pre-testing the update via CI, with the caveat that it may or may not timeout. We shall see.

mascguy commented:
If anyone is interested in pre-testing this locally - without having to pull from my branch - that's now possible via port `libgit2-devel`.

As of last week, all dependents now use a path-style dep, so that port can serve as a drop-in replacement. Done via commit: https://github.com/macports/macports-ports/commit/22b4db7eb8beafdc330f0d6e04bd61bd856ef745

So if anyone's running a speedy Mac - and you don't mind rebuilding Rust, Cargo, etc - it would be great to have a few more points of reference...

mascguy commented:
Dropped rev-bumps for `mergestat` and `geany-plugins`, as both ports are broken across-the-board.

Ticket for former:
- https://trac.macports.org/ticket/65756

Ticket for latter:
- https://trac.macports.org/ticket/67826

barracuda156 commented:
@mascguy I will try now Fortran and R dependents with `libgit2-devel`.

mascguy commented on devel/libgit2/Portfile:
```diff
@@ -11,7 +11,7 @@ conflicts           libgit2-devel
-github.setup        libgit2 libgit2 1.6.4 v
+github.setup        libgit2 libgit2 1.7.0 v
 github.tarball_from archive
 revision            1
```
I failed to set the revision back to zero, but will take care of before merging.

mascguy replied:
Revision fixed

mascguy commented:
Unfortunately it looks like `py-git2` is not yet ready `libgit2` 1.7.0. Upstream is working on it, with the following commit covering the changes made thus far:

https://github.com/libgit2/pygit2/commit/1473e8eb6eb59dc7521dcd5f8a4c9390e9b53223

However, we'll want to wait for a new formal release. So we need to remain with v1.6.4 for now.

mascguy commented:
> Unfortunately it looks like `py-git2` is not yet ready `libgit2` 1.7.0. Upstream is working on it, with the following commit covering the changes made thus far:
> 
> [libgit2/pygit2@1473e8e](https://github.com/libgit2/pygit2/commit/1473e8eb6eb59dc7521dcd5f8a4c9390e9b53223)
> 
> However, we'll want to wait for a new formal release. So we need to remain with v1.6.4 for now.

This assumes that our critical toolchain components like `cargo`/`cargo-c`, `rust`, etc, are compatible with v1.7.0. If not, then we'll be waiting longer.

In short, we need to wait for both foundational components, as well as the vast majority of the dependents, to be compatible. Sort of goes without saying, but just to state the obvious in case anyone's wondering...

barracuda156 commented:
@mascguy By the way, `libgit2` 1.6.4 now fails to build with `legacy-support-devel` 20230722:
```
make[2]: Entering directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build'
[  0%] Building C object src/libgit2/CMakeFiles/libgit2.dir/annotated_commit.c.o
cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build/src/libgit2 && /usr/bin/gcc-4.2  -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build/src/util -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build/include -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2 -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/include -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/deps/http-parser -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/deps/ntlmclient -isystem /opt/local/libexec/openssl3/include -isystem /opt/local/include -D_GNU_SOURCE -pipe -Os -I/opt/local/include/LegacySupport -DNDEBUG -I/opt/local/include -I/opt/local/include/LegacySupport -Werror -Wno-error -Wall -Wextra -fvisibility=hidden -fPIC -Wno-missing-field-initializers -Wmissing-declarations -Wstrict-aliasing -Wstrict-prototypes -Wdeclaration-after-statement -Wunused-function -Wformat -Wformat-security -O2 -g -DNDEBUG -std=c89 -arch ppc -mmacosx-version-min=10.6 -MD -MT src/libgit2/CMakeFiles/libgit2.dir/annotated_commit.c.o -MF CMakeFiles/libgit2.dir/annotated_commit.c.o.d -o CMakeFiles/libgit2.dir/annotated_commit.c.o -c /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/annotated_commit.c
In file included from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util/unix/posix.h:13,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util/posix.h:157,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util/git2_util.h:106,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/common.h:10,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/annotated_commit.h:10,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/annotated_commit.c:8:
/opt/local/include/LegacySupport/dirent.h:60: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘struct’
/opt/local/include/LegacySupport/dirent.h:64: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘int’
/opt/local/include/LegacySupport/dirent.h:68: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘void’
/opt/local/include/LegacySupport/dirent.h:72: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘void’
/opt/local/include/LegacySupport/dirent.h:76: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘long’
make[2]: *** [src/libgit2/CMakeFiles/libgit2.dir/annotated_commit.c.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build'
make[1]: *** [src/libgit2/CMakeFiles/libgit2.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
Also with gcc12:
```
[  0%] Building C object src/libgit2/CMakeFiles/libgit2.dir/annotated_commit.c.o
cd /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build/src/libgit2 && /opt/local/bin/gcc-mp-12  -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build/src/util -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/build/include -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2 -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/include -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/deps/http-parser -I/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/deps/ntlmclient -isystem /opt/local/libexec/openssl3/include -isystem /opt/local/include -D_GNU_SOURCE -pipe -Os -I/opt/local/include/LegacySupport -DNDEBUG -I/opt/local/include -I/opt/local/include/LegacySupport -Werror -Wno-error -Wall -Wextra -fvisibility=hidden -fPIC -Wno-documentation-deprecated-sync -Wno-missing-field-initializers -Wmissing-declarations -Wstrict-aliasing -Wstrict-prototypes -Wdeclaration-after-statement -Wshift-count-overflow -Wunused-const-variable -Wunused-function -Wint-conversion -Wc99-c11-compat -Wformat -Wformat-security -O2 -g -DNDEBUG -std=c90 -arch ppc -mmacosx-version-min=10.6 -MD -MT src/libgit2/CMakeFiles/libgit2.dir/annotated_commit.c.o -MF CMakeFiles/libgit2.dir/annotated_commit.c.o.d -o CMakeFiles/libgit2.dir/annotated_commit.c.o -c /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/annotated_commit.c
In file included from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util/unix/posix.h:13,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util/posix.h:157,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/util/git2_util.h:106,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/common.h:10,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/annotated_commit.h:10,
                 from /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_libgit2/libgit2/work/libgit2-1.6.4/src/libgit2/annotated_commit.c:8:
/opt/local/include/LegacySupport/dirent.h:60:14: error: expected ';' before 'struct'
   60 | static inline struct dirent *__mpls_readdir(DIR *dir) {
      |              ^~~~~~~
      |              ;
/opt/local/include/LegacySupport/dirent.h:64:14: error: expected ';' before 'int'
   64 | static inline int __mpls_readdir_r(DIR *dir, struct dirent *entry, struct dirent **result) {
      |              ^~~~
      |              ;
/opt/local/include/LegacySupport/dirent.h:68:14: error: expected ';' before 'void'
   68 | static inline void __mpls_rewinddir(DIR *dir) {
      |              ^~~~~
      |              ;
/opt/local/include/LegacySupport/dirent.h:72:14: error: expected ';' before 'void'
   72 | static inline void __mpls_seekdir(DIR *dir, long loc) {
      |              ^~~~~
      |              ;
/opt/local/include/LegacySupport/dirent.h:76:14: error: expected ';' before 'long'
   76 | static inline long __mpls_telldir(DIR *dir) {
      |              ^~~~~
      |              ;
```
Possibly/hopefully will be fixed by: https://github.com/macports/macports-legacy-support/pull/62

macportsraf commented:
I could be wrong but it looks like maybe the compiler isn't recognizing `inline` as intended. I've changed https://github.com/macports/macports-legacy-support/pull/62 so that the inline functions are no longer inline. Maybe that'll help.

barracuda156 commented:
@macportsraf libgit2 builds fine with legacy-support-devel from 8889f27039e355c2ccadda0240ddd540302e382f

macportsraf commented:
@barracuda156 Thanks! That's great news.

mascguy commented:
Observers: We're still waiting on an update to `pygit2`, so this is still blocked for now.

Nonetheless, upstream just released a new version of `libgit2` - 1.7.1 - so I've updated this PR with that.

mascguy commented:
Update: It looks like upstream is making progress on updating `pygit2`, based on recent commit history. But still waiting for an official release.

mascguy commented:
> Update: It looks like upstream is making progress on updating `pygit2`, based on recent commit history. But still waiting for an official release.

`pygit2` has now been updated to support `libgit2`, as of release 1.13. Added that update to this PR, rebased, and removed `[skip ci]`.

Let's see if everything builds!

mascguy commented:
All, CI looks good, so this is good to merge. However, let's wait until the buildbots catch up with their current backlog. (Since this PR includes rebuilds for `rust` and `cargo`, among others, that will involve many more hours of builds.)
